### ISSUE TITLE
 Request for Cypress Open Support in NX v18 Inferred Targets

### ISSUE THREAD

**Issue by ali0120** [nrwl/nx#21857]
### Current Behavior

NX v18's inferred targets (part of "Project Crystal") currently utilizes the Cypress CLI, lacking support for Cypress Open. This means users are unable to interact with tests in watch mode, hindering development efficiency. Requesting reinstatement of Cypress Open support to enhance developer experience within inferred targets. Opening issue in NX repo for resolution. Thanks!

### Expected Behavior

I expect that NX v18 inferred targets should support the Cypress Open command, allowing users to interact with tests in watch mode. This would enable a seamless testing experience and enhance developer productivity within NX projects.

### Steps to Reproduce

Run the command nx run header:component-test --watch in an NX v18 project with inferred targets enabled.
Observe that the command executes without error but does not provide the expected behavior of launching Cypress in watch mode.
Notice that despite using the --watch flag, Cypress does not open in watch mode, preventing users from interacting with tests interactively.
Attempt to use the --headed flag as a workaround to observe test execution, but realize that it does not provide the same interactive experience as Cypress Open.
Confirm the issue persists even after ensuring that all dependencies and configurations are correctly set up.
Compare the behavior with previous versions of NX or non-inferred target setups to verify that the issue specifically occurs within NX v18 inferred targets.
Document any additional observations or troubleshooting steps taken to isolate the issue.

### Nx Report

```shell
With params --watch
No terminal output was found for this task. Nx only sends the terminal output for cacheable operations.
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by pumano** [comment]
I use command like: `nx e2e onboarding-e2e --base-url=http://localhost:4200/ibc/onboarding --headed --browser chrome --watch` and it works perfectly, but when I migrate to crystal plugin it's not working, I need somehow manually run cypress.

**Comment by leosvelperez** [comment]
This was addressed by https://github.com/nrwl/nx/pull/22556 and released in [Nx 18.2.0](https://github.com/nrwl/nx/releases/tag/18.2.0). The Cypress plugin will now [infer an `open-cypress` task](https://nx.dev/nx-api/cypress/documents/overview#watching-for-changes-headed-mode) that runs `cypress open`.
